api/v1alpha1/stage_types.go

```diff
@@ -567,6 +576,60 @@ type FreightReference struct {
 	VerificationHistory VerificationInfoStack `json:"verificationHistory,omitempty" protobuf:"bytes,7,rep,name=verificationHistory"`
 }
 
+// FreightSelection is a collection of FreightReferences, each of which
+// represents a piece of Freight that has been selected for deployment to a
+// Stage.
+type FreightSelection struct {
+	// Items is a map of FreightReference objects, indexed by their Warehouse
+	// origin.
+	Items map[string]FreightReference `json:"items,omitempty" protobuf:"bytes,1,rep,name=items" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
```

Comment:
This was the compromise required to get this to work with protobufs, right? I am thinking we should add some note about it here.

Reply:
I thought about this, but opted not to because it would leak into the CRD documentation. Where this level of detail may end up confusing the user.

Reply:
Fair enough. I have sometimes wished there were a straightforward way to leave type or field-level comments that _don't_ bleed into those docs.

Reply:
Commit messages are the answer, but this does not really pair well with a squash and merge strategy.